### PR TITLE
ETFE-3144 CAM-NEE13 EORI new page added

### PR DESCRIPTION
**Pull request by grahampaulcook**

### CODE REVIEW

**Comment by grahampaulcook** [comment]
There is an outstanding query on the validation with UX

**Review by LewisMitchell — CHANGES_REQUESTED** [review]

File: app/viewmodels/checkAnswers/sections/consignee/ConsigneeExportEoriSummary.scala [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+          actions = if (!showActionLinks) Seq() else Seq(
+            ActionItemViewModel(
+              content = "site.change",
+              href = routes.ConsigneeExportEoriController.onPageLoad(request.ern, request.draftId, CheckMode).url,
+              id = "changeConsigneeExportEori"
+            ).withVisuallyHiddenText(messages("consigneeExportEori.change.hidden"))
+          )
+        )
+    }
+}
```
Comment (LewisMitchell) [inline app/viewmodels/checkAnswers/sections/consignee/ConsigneeExportEoriSummary.scala:47]: Could you add some tests for this please?

File: app/forms/sections/consignee/ConsigneeExportEoriFormProvider.scala [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+
+import javax.inject.Inject
+
+class ConsigneeExportEoriFormProvider @Inject() extends Mappings {
+
+  private val EORI_NUMBER_MAX_LENGTH = 17
+
+  def apply(): Form[String] =
+    Form(
+      "value" -> text("consigneeExportEori.error.required")
```
[the remark this answers is not in the document]
Comment (grahampaulcook) [reply app/forms/sections/consignee/ConsigneeExportEoriFormProvider.scala:31]: I didn't see that as a requirement and it's not currently being done at https://github.com/hmrc/emcs-tfe-create-movement-frontend/blob/main/app/forms/sections/consignee/ConsigneeExportInformationFormProvider.scala#L48

File: test/forms/sections/consignee/ConsigneeExportEoriFormProviderSpec.scala [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+    behave like mandatoryField(
+      form,
+      fieldName,
+      requiredError = FormError(fieldName, requiredKey)
+    )
+
+    s"not bind strings longer than 17 in length" in {
+      val veryLongString = "123456789012345678"
+      val result = form.bind(Map(fieldName -> veryLongString)).apply(fieldName)
+      result.errors must contain only FormError(fieldName, lengthKey, Seq(maxLength))
```
[the remark this answers is not in the document]
Comment (grahampaulcook) [reply test/forms/sections/consignee/ConsigneeExportEoriFormProviderSpec.scala]: Updated to use `fieldWithMaxLength` on https://github.com/hmrc/emcs-tfe-create-movement-frontend/pull/245/commits/857dfb4c523ef1095bd23901d8b7a20aefb654d3

File: app/viewmodels/checkAnswers/sections/consignee/ConsigneeExportEoriSummary.scala (hunk 1 above)
Comment (grahampaulcook) [reply app/viewmodels/checkAnswers/sections/consignee/ConsigneeExportEoriSummary.scala:47]: Added missing tests https://github.com/hmrc/emcs-tfe-create-movement-frontend/pull/245/commits/46a138e7450b99004dcef7a3623cd0257426371a

**Review by LewisMitchell — APPROVED** [review]

**Pull request merged**